plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1151,6 +1166,128 @@ private void testCreateTableColumnMappingMode(ColumnMappingMode mode, Consumer<S
+
+        assertUpdate("CREATE TABLE " + tableName + " (a_number INT, b_number INT) WITH (column_mapping_mode='" + mode + "')");
+
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1, 10), (2, 20), (null, null)", 3);
+        assertUpdate("ANALYZE " + tableName);
+
+        assertQuery("SHOW STATS FOR " + tableName, "VALUES" +
+                "('a_number', null, 2.0, 0.33333333333, null, 1, 2)," +
+                "('b_number', null, 2.0, 0.33333333333, null, 10, 20)," +
+                "(null, null, null, null, 3.0, null, null)");
+
+        // Ensure SHOW STATS return the same stats for the restored column
+        assertUpdate("ALTER TABLE " + tableName + " RENAME COLUMN b_number TO new_b");
+        assertQuery("SHOW STATS FOR " + tableName, "VALUES" +
+                "('a_number', null, 2.0, 0.33333333333, null, 1, 2)," +
+                "('new_b', null, 2.0, 0.33333333333, null, 10, 20)," +
+                "(null, null, null, null, 3.0, null, null)");
+
+        // Re-analyzing should work
+        assertUpdate("ANALYZE " + tableName);
+        assertQuery("SHOW STATS FOR " + tableName, "VALUES" +
+                "('a_number', null, 2.0, 0.33333333333, null, 1, 2)," +
+                "('new_b', null, 2.0, 0.33333333333, null, 10, 20)," +
+                "(null, null, null, null, 3.0, null, null)");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test(dataProvider = "columnMappingModeDataProvider")
+    public void testCommentOnTableForColumnMappingMode(ColumnMappingMode mode)
+    {
+        String tableName = "test_comment_on_table_for_column_mapping_mode_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (a_number INT, b_number INT) WITH (column_mapping_mode='" + mode + "')");
+
+        assertUpdate("COMMENT ON TABLE " + tableName + " IS 'test comment' ");
+        assertQuery("SELECT comment FROM system.metadata.table_comments WHERE table_name = '" + tableName + "'", "VALUES ('test comment')");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test(dataProvider = "columnMappingModeDataProvider")
+    public void testCommentOnColumnForColumnMappingMode(ColumnMappingMode mode)
+    {
+        String tableName = "test_comment_on_column_for_column_mapping_mode_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (a_number INT, b_number INT) WITH (column_mapping_mode='" + mode + "')");
+
+        assertUpdate("COMMENT ON COLUMN " + tableName + ".a_number IS 'test column comment'");
+        assertQuery("SELECT comment FROM delta.information_schema.columns WHERE table_name = '" + tableName + "' AND column_name = 'a_number'", "VALUES ('test column comment')");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test(dataProvider = "columnMappingModeDataProvider")
+    public void testCreateTableWithCommentsColumnForColumnMappingMode(ColumnMappingMode mode)
```

Comment:
nit: I would rename to `testCreateTableWithCommentsForColumnMappingMode` (remove Column) because this test contains a table comment.